### PR TITLE
Use Typescript

### PR DESCRIPTION
**Pull request by thewilkybarkid**
Introduces Node.js, NPM and TypeScript.

Requires https://github.com/libero/article-page/pull/1.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+Atomics.wait(new Int32Array(new SharedArrayBuffer(4)), 0, 0, 864000);
```
Comment (giorgiosironi) [inline src/index.ts:2]: That seems complicated, but temporary; comment saying it's a sleep?

File: tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,12 @@
+{
```
Comment (giorgiosironi) [inline tsconfig.json:1]: #cop-js people required here

Comment (thewilkybarkid) [reply tsconfig.json:1]: Matches https://github.com/libero/article-store/blob/master/tsconfig.json.

File: tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+  "compilerOptions": {
+    "target": "es2019",
+    "module": "commonjs",
+    "outDir": "build",
+    "strict": true,
```
Comment (davidcmoulton) [inline tsconfig.json:6]: Does this cover `strictNullChecks` or is that separate? If separate, add it in?

File: README.md [hunk 4]
```diff
@@ -50,6 +53,21 @@ make dev
 
 The application will then sleep for 24 hours.
 
+<details>
+
+<summary>Rebuilding the container</summary>
+
+Code is attached to the containers as volumes so most updates are visible without a need to rebuild the container.
+However, changes to NPM dependencies, for example, require a rebuild. So you may need to execute
```
Comment (davidcmoulton) [inline README.md:61]: Worth having make targets that hide this complexity?

File: tsconfig.json (hunk 3 above)
Comment (thewilkybarkid) [reply tsconfig.json:6]: > Enable all strict type checking options.
> Enabling `--strict` enables `--noImplicitAny`, `--noImplicitThis`, `--alwaysStrict`, `--strictBindCallApply`, `--strictNullChecks`, `--strictFunctionTypes` and `--strictPropertyInitialization`.

💪

File: README.md (hunk 4 above)
Comment (thewilkybarkid) [reply README.md:61]: I haven't been making it a dependency as a `build` is not needed in most cases. (Due to Docker caches it's not that slow, but still...)

**Review by giorgiosironi — APPROVED** [review]
No more concerns, but I'm not an authority on TypeScript

**Pull request merged**